frontend/util.lua

```diff
@@ -763,15 +763,33 @@ end
 -- @string path the directory to create
 -- @treturn bool true on success; nil, err_message on error
 function util.makePath(path)
-    path = path:gsub("/+$", "")
-    if util.pathExists(path) then return true end
+    if lfs.attributes(path, "mode") == "directory" then
+        return true
+    end
```

Comment:
Hm, what if it's a file?

Reply:
It'll go through and fail via mkdir ;).

Reply:
Specifically:

```
> print(makePath("baz"))
nil     File exists (creating `baz/` for `baz`)
> print(makePath("baz/"))
nil     File exists (creating `baz/` for `baz/`)
```

Reply:
Feels potentially a bit late/unclear if we already know right there that if ~= "directory" and ~= nil then danger.

Reply:
That saves us from writing a dedicated early trap, and allows it to work no matter where the spurious file is in the tree (e.g., foo/bar/baz if bar is the file).

Reply:
It also automagically handles the "trailing slash or not" conundrum, as `foo` == `foo/` for `mkdir`, but not necessarily for `stat`:

```
> print(lfs.attributes("baz", "mode"))
file
> print(lfs.attributes("baz/", "mode"))
nil     cannot obtain information from file 'baz/': Not a directory     20
```